extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/QuarkusExecutorFactory.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.vertx.core.runtime;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadFactory;
+
+import org.jboss.threads.EnhancedQueueExecutor;
+import org.jboss.threads.JBossExecutors;
+import org.wildfly.common.cpu.ProcessorInfo;
+
+import io.quarkus.vertx.core.runtime.config.VertxConfiguration;
+import io.vertx.core.spi.ExecutorServiceFactory;
+
+public class QuarkusExecutorFactory implements ExecutorServiceFactory {
+    private final VertxConfiguration conf;
+
+    public QuarkusExecutorFactory(VertxConfiguration conf) {
+        this.conf = conf;
+    }
+
+    @Override
+    public ExecutorService createExecutor(ThreadFactory threadFactory, Integer concurrency, Integer maxConcurrency) {
+        final EnhancedQueueExecutor.Builder builder = new EnhancedQueueExecutor.Builder()
```

Comment:
Can we just re-use the existing executor from ExecutorRecorder here?

My thinking is:
- Deprecate the config that sets the vert.x blocking thread pool size. If this is explicity set then use it, but print a warning
- Use Integer.MIN_VALUE to represent 'use the existing pool'. Pass this into the factory to tell it to just re-use the existing factory (this may get caught by validation, we may have to use MAX_VALUE instead)
- Allow the internal pool for file operations to be created as normal

Reply:
If the vert.x blocking thread pool size is set, by "use it" do you mean create a new EQE instance? Otherwise, I'm not sure how we can have the vert.x thread pool config override the overall quarkus thread pool config in _core_?

Right now all three pools in Vert.x use the same factory, so there's no way to separate out the regular blocking pool executor factory and shared executor pool, from the file operations one.

Reply:
I guess you *could* return the same executor for all three.  That might introduce problems though if Vert.x is specifically designed to assume that certain thread pools never overlap.

Reply:
I think you can merge the two main blocking ones, and leave the vertx-internal one as a separate pool. vertx-internal is only used for non blocking file operations (so super short tasks), and mostly will never be used.

The separation between the other two is kinda historical/based on technical limitations, there should be no issues with the vert.x pool being our main executor pool.

Reply:
`VertxImpl` creates three separate pools using the same executor factory:
- https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/impl/VertxImpl.java#L157
- https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/impl/VertxImpl.java#L160
- https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/impl/VertxImpl.java#L1122

I don't think there's any way for us to know when we're creating an executor for blocking file operations vs anything else. At least without modifying Vert.x.

Are you saying we shouldn't be creating a separate `ExecutorFactory` for each call to the factory, but instead just pass out the EQE instance we create in Quarkus core? Essentially having a single thread pool for anything not running on the event loop for everything in Quarkus?

Reply:
> Are you saying we shouldn't be creating a separate `ExecutorFactory` for each call to the factory, but instead just pass out the EQE instance we create in Quarkus core? Essentially having a single thread pool for anything not running on the event loop for everything in Quarkus?

I was definitely *not* saying we *should* (and I'm not saying we *should not*), but I am saying that this is *possible*.

Another thing that is possible: they way they've implemented this means that we *could* keep a simple counter in our executor factory and know which pool is being created by its sequence.  We could decide which pool to create or return based on this information.  Ugly but effective.

Reply:
Apologies, I should've clarified I was directing that particular comment at @stuartwdouglas.

I'm fine with whatever approach we want to take, but I need to know which way to "jump" so I can finalize the PR and continue with the changes for OpenTelemetry

Reply:
Ah, sorry about that, I wasn't sure.

Reply:
Not a problem, and no need to be sorry.

Reply:
https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/impl/VertxImpl.java#L157 is the one that can be shared. We can make the factory know which one it is creating through the use of a 'magic value' (likely Integer.MAX_VALUE, as we can't easily pass in a negative number). 

Conceptually this pool and the ExecutorRecorder pool perform the same task, so I don't really see any reason to have the two pools.

Reply:
@stuartwdouglas by using a magic number, do you mean setting Vert.x configuration for those pools that aren't shared to MAX_VALUE but then in the executor actually set it to the same value we have for the other pool config?